### PR TITLE
Updates to README

### PR DESCRIPTION
**Pull request by hjdivad**
This is a first pass at docs but should be a reasonable starting point.

### CODE REVIEW

**Comment by hjdivad** [comment]
No need for extremely detailed scrutiny but definitely happy to get feedback (on all of `README` not just the diff in this pr)

File: README.md [hunk 1]
```diff
@@ -208,36 +216,349 @@ normalization to JSON API.
...
+
+Because models (or `RecordArray`s of models are cached) the cache can be emptied
+automatically when the models are unloaded.  In the case of `RecordArray`s of
+models, the entire cache entry is evicted if *any* of the member models is
+unloaded.
+
 
 ## Schema
 
+You have to register a schema to tell ember-m3 what types it should be enabled
```
Comment (stefanpenner) [inline README.md]: this should include, what is a schema, and when it is appropriate to use one, and how they differ from models.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# ember-m3 [![Build Status](https://secure.travis-ci.org/hjdivad/ember-m3.svg?branch=master)](http://travis-ci.org/hjdivad/ember-m3)
+# Ember-m3 [![Build Status](https://secure.travis-ci.org/hjdivad/ember-m3.svg?branch=master)](http://travis-ci.org/hjdivad/ember-m3)
```
Comment (hjdivad) [inline README.md]: @rwjblue I could go either way.  I take it you're in favour of always lowercase (even in title / start of sentence)?

Comment (rwjblue) [inline README.md]: I think the only one dislike is what is here ATM (`Ember-m3`). 

IMHO, if we are referring to "the package" we should always use the package's actual name on NPM (which is `ember-m3`).  I personally would be happy to _only_ use `ember-m3`, but could also see where folks might prefer to use `Ember M3` when discussing things in prose.

Feel free to completely disregard this entire comment though 😝

Comment (hjdivad) [inline README.md]: Nope I'm convinced; I'll switch everything to `ember-m3` and consider `Ember m3` or `Ember M3` documentation bugs.

File: README.md [hunk 3]
```diff
@@ -208,36 +216,349 @@ normalization to JSON API.
...
+}
+```
+
+
+### Other Computed Properties
+
+More involved computed properites can be converted to either utility functions
+(if used within JavaScript) or helper functions (if used in templates).
+
+For properties used in both templates and elsehwere (eg components) a convenient
```
Comment (jsturgis) [inline README.md]: elsehwere -> elsewhere

Comment (hjdivad) [reply README.md]: thanks @jsturgis !

**Pull request merged**